#  Morphological data analysis
## Plots for morphological data

### PCA Plots combining all the data for both species *S. maderensis* & *S. aurita*

![plot1](https://github.com/evelyn-takyi/Sardinella-project/blob/master/Aurita_mad_plot1.png)
![plot2](https://github.com/evelyn-takyi/Sardinella-project/blob/master/Aurita_mad_plot2.png)

The morphological data is not able to show differentiation between *S.aurita* and *S.maderensis*

## *S.aurita* species

### Morphological Data

| ID   | Population   | Plate name   | FK   | G    | W     | TL   |
|------|--------------|--------------|------|------|-------|------|
| GH1  | Ghana.A      | A1           | 15   | 9    | 43.2  | 20   |
| GH4  | Ghana.A      | A4           | 16.5 | 9    | 53.6  | 17   |
| GH2  | Ghana.A      | A2           | 15   | 8.5  | 45.9  | 18   |
| GH5  | Ghana.A      | A5           | 15   | 8.5  | 44.3  | 19   |
| GH6  | Ghana.A      | A6           | 14.5 | 8    | 44.2  | 17   |
| GH7  | Ghana.A      | A7           | 17.5 | 9    | 59.4  | 17   |
| GH8  | Ghana.A      | A8           | 15   | 8.5  | 46.4  | 19.6 |
| GH9  | Ghana.A      | A9           | 16   | 8.5  | 41.2  | 17.4 |
| GH10 | Ghana.A      | A10          | 16.5 | 9.5  | 55.8  | 18   |
| GH5  | Ghana.A      | F5           | 12   | 10   | 44.3  | 19   |
| GH11 | Ghana.A      | F6           | 11.5 | 6    | 17.6  | 13   |
| GH12 | Ghana.A      | F7           | 11.5 | 6    | 17.6  | 13   |
| GH13 | Ghana.A      | F8           | 11   | 6    | 15.7  | 12.6 |
| GH14 | Ghana.A      | F9           | 11.5 | 6    | 16.1  | 13   |
| GH15 | Ghana.A      | F10          | 12   | 6    | 15.3  | 12.5 |
| GH43 | Ghana.A      | B1           | 18   | 10   | 23.6  | 20.5 |
| GH44 | Ghana.A      | B2           | 17   | 10   | 71.9  | 19.6 |
| GH45 | Ghana.A      | B3           | 17   | 9.5  | 68.3  | 20.5 |
| GH46 | Ghana.A      | B4           | 17   | 10   | 66.7  | 20   |
| GH47 | Ghana.A      | B5           | 17   | 10   | 60.3  | 19   |
| T2   | Togo.A       | 2E           | 15   | 8.5  | 45    | 17.5 |
| T3   | Togo.A       | 3E           | 17   | 9.5  | 68    | 17   |
| T4   | Togo.A       | 4E           | 16   | 9.5  | 62    | 19   |
| T5   | Togo.A       | 5E           | 15.5 | 9    | 53    | 19   |
| T6   | Togo.A       | 6E           | 15   | 8    | 41    | 18   |
| T7   | Togo.A       | 7E           | 15.5 | 8    | 50    | 17   |
| T8   | Togo.A       | 8E           | 15   | 9    | 48    | 17.5 |
| T9   | Togo.A       | 9E           | 16   | 10   | 61    | 17.5 |
| T10  | Togo.A       | 10E          | 15   | 9    | 40    | 19   |
| T41  | Togo.A       | T41          | 18   | 10   | 82    | 17   |
| T44  | Togo.A       | T44          | 16   | 9    | 60    | 20.5 |
| T45  | Togo.A       | T45          | 16.5 | 8    | 60    | 18.5 |
| T47  | Togo.A       | T47          | 14.5 | 9    | 40    | 16.5 |
| T21  | Togo.A       | 1D           | 17.5 | 10.5 | 70    | 17.5 |
| T22  | Togo.A       | 2D           | 16.5 | 9    | 61    | 20   |
| T23  | Togo.A       | 3D           | 16.5 | 8    | 38    | 19   |
| T24  | Togo.A       | 4D           | 16   | 9    | 59    | 16.5 |
| T25  | Togo.A       | 5D           | 16   | 8.5  | 52    | 19   |
| T26  | Togo.A       | 6D           | 17.5 | 8    | 47    | 18.5 |
| M70  | Mauritania.A | D2           | 26.4 | 8.3  | 306   | 0    |
| M74  | Mauritania.A | D3           | 26.1 | 8.1  | 304   | 0    |
| M66  | Mauritania.A | D5           | 28.5 | 8.2  | 360   | 0    |
| M60  | Mauritania.A | D7           | 27.3 | 8.2  | 330   | 0    |
| M58  | Mauritania.A | D8           | 27.5 | 8    | 330   | 0    |
| M41  | Mauritania.A | D9           | 32.5 | 9.5  | 527   | 0    |
| M1   | Mauritania.A | G1           | 24   | 8.4  | 231   | 0    |
| M2   | Mauritania.A | G2           | 24.7 | 8.3  | 259   | 0    |
| M3   | Mauritania.A | G3           | 23.9 | 6.5  | 235   | 0    |
| M4   | Mauritania.A | G4           | 24.5 | 8    | 258   | 0    |
| M5   | Mauritania.A | G5           | 25.2 | 8.5  | 252   | 0    |
| M6   | Mauritania.A | G6           | 22.5 | 7.5  | 202   | 0    |
| M8   | Mauritania.A | G8           | 25.2 | 8.5  | 269   | 0    |
| M10  | Mauritania.A | G10          | 26.6 | 9    | 311   | 0    |
| M50  | Mauritania.A | 1A           | 33   | 8.5  | 510   | 0    |
| M58  | Mauritania.A | 2A           | 27.5 | 8    | 330   | 0    |
| M59  | Mauritania.A | 7A           | 26.8 | 8.5  | 320   | 0    |
| M60  | Mauritania.A | 8A           | 27.3 | 8.2  | 330   | 0    |
| M66  | Mauritania.A | 9A           | 28.5 | 8.2  | 360   | 0    |
| S1   | Senegal.A    | C1           | 25.5 | 6.2  | 260.7 | 30.8 |
| S2   | Senegal.A    | C2           | 27.5 | 6    | 296.7 | 32.8 |
| S3   | Senegal.A    | C3           | 26.5 | 6.2  | 245.5 | 31.5 |
| S4   | Senegal.A    | C4           | 25.1 | 6.5  | 243.3 | 30.7 |
| S5   | Senegal.A    | C5           | 25.7 | 6    | 246.2 | 31.1 |
| S6   | Senegal.A    | C6           | 26.5 | 6.5  | 281.3 | 31.8 |
| S7   | Senegal.A    | C7           | 26.7 | 6    | 252.7 | 32   |
| S8   | Senegal.A    | C8           | 26.8 | 6.5  | 231.9 | 31.5 |
| S9   | Senegal.A    | C9           | 27.4 | 6.5  | 278.8 | 32.5 |
| S10  | Senegal.A    | C10          | 26.7 | 6.1  | 291.6 | 32.3 |
| S21  | Senegal.A    | IC           | 25   | 6.3  | 302.1 | 32.1 |
| S22  | Senegal.A    | 2C           | 25.5 | 6.5  | 251.4 | 31   |
| S23  | Senegal.A    | 3C           | 27.2 | 6.3  | 285.2 | 32.4 |
| S25  | Senegal.A    | 5C           | 27   | 5.7  | 281.4 | 32.5 |
| S27  | Senegal.A    | 7C           | 25.7 | 6    | 210.8 | 30.2 |
| S28  | Senegal.A    | 8C           | 25   | 6.5  | 231.2 | 30   |
| S38  | Senegal.A    | S38          | 25.3 | 5.7  | 216.6 | 28.1 |
| S24  | Senegal.A    | 4C           | 26.2 | 6.4  | 259   | 31.8 |
| G1   | Guinea.A     | Guinea1_New  | 17   | 4    | 61.1  | 20   |
| G11  | Guinea.A     | Guinea11_New | 15   | 4.5  | 47.1  | 17.5 |
| G13  | Guinea.A     | Guinea13_New | 15.5 | 4.5  | 53.5  | 18.5 |
| G14  | Guinea.A     | Guinea14_New | 18   | 4    | 83    | 20   |
| G15  | Guinea.A     | Guinea15_New | 16.5 | 4    | 52.3  | 19   |
| G16  | Guinea.A     | Guinea16_New | 17   | 4.5  | 59.5  | 19.5 |
| G19  | Guinea.A     | Guinea19_New | 16   | 5    | 51.1  | 19   |
| G2   | Guinea.A     | Guinea2_New  | 16.5 | 2.8  | 57.4  | 20   |
| G21  | Guinea.A     | Guinea21_New | 17   | 4    | 65.5  | 21   |
| G22  | Guinea.A     | Guinea22_New | 19   | 5    | 78.1  | 22.5 |
| G23  | Guinea.A     | Guinea23_New | 16   | 5    | 56.3  | 18   |
| G25  | Guinea.A     | Guinea25_New | 18   | 5    | 69.9  | 18.5 |
| G31  | Guinea.A     | Guinea31_New | 18   | 5    | 59    | 19   |
| G37  | Guinea.A     | Guinea37_New | 16   | 4    | 59    | 21.5 |
| G43  | Guinea.A     | Guinea43_New | 16   | 4.5  | 56.2  | 19.5 |
| G1   | Guinea.A     | Guinea_11G   | 17   | 4    | 61.1  | 20   |
| G2   | Guinea.A     | Guinea_1G    | 16.5 | 2.8  | 57.4  | 20   |
| G3   | Guinea.A     | Guinea_1H    | 16   | 4    | 56.4  | 19.5 |
| G4   | Guinea.A     | Guinea_4G    | 16.5 | 5    | 63.5  | 20   |
| G5   | Guinea.A     | Guinea_5H    | 16   | 4.5  | 57.2  | 20   |

###  PCA Plots of morphological data for *S. aurita*

```library(ggfortify)
df <- data[c(4,5,6)]
MYDATA <- autoplot(prcomp(df))

autoplot(prcomp(df), data = data, colour = 'Population')

autoplot(prcomp(df), data = data, colour = 'Population', label=TRUE, shape = FALSE, label.size = 3, legend=F)

```
![plot1](https://github.com/evelyn-takyi/Sardinella-project/blob/master/Aurita_plot2.png)

![plot2](https://github.com/evelyn-takyi/Sardinella-project/blob/master/Aurita_plot1.png)


## *S. maderensis*

### Morphological Data


| ID  | Population | Taxon        | FK   | G    | W     | TL   |
|-----|------------|--------------|------|------|-------|------|
| M7  | G7         | Mauritania.M | 25.5 | 8.7  | 280   | 0    |
| M9  | G9         | Mauritania.M | 26.9 | 8.6  | 321   | 0    |
| M75 | D4         | Mauritania.M | 27.8 | 8.4  | 352   | 0    |
| M50 | D6         | Mauritania.M | 33   | 8.5  | 510   | 0    |
| M42 | D10        | Mauritania.M | 32.5 | 9.5  | 515   | 0    |
| M79 | 1A         | Mauritania.M | 25.7 | 8.3  | 286   | 0    |
| M80 | 2A         | Mauritania.M | 22.6 | 7.2  | 201   | 0    |
| M88 | 4A         | Mauritania.M | 22.5 | 7.1  | 185   | 0    |
| M89 | 5A         | Mauritania.M | 24.6 | 8.5  | 235   | 0    |
| M9  | 6A         | Mauritania.M | 26.9 | 8.6  | 321   | 0    |
| M25 | 7A         | Mauritania.M | 24.8 | 8.4  | 251   | 0    |
| M90 | 7B         | Mauritania.M | 25.4 | 8.3  | 275   | 0    |
| M98 | 8A         | Mauritania.M | 23.3 | 7.7  | 214   | 0    |
| M16 | 9A         | Mauritania.M | 24.5 | 8.6  | 244   | 0    |
| M99 | 9B         | Mauritania.M | 26.7 | 8.7  | 304   | 0    |
| M17 | 10A        | Mauritania.M | 25.3 | 7.6  | 265   | 0    |
| M14 | 11A        | Mauritania.M | 25.9 | 7.5  | 281   | 0    |
| M12 | 11B        | Mauritania.M | 26.5 | 8.6  | 298   | 0    |
| M85 | 12A        | Mauritania.M | 22.3 | 7.1  | 203   | 0    |
| S1  | H1         | Senegal.M    | 24   | 6.6  | 239.4 | 29.5 |
| S2  | H2         | Senegal.M    | 25   | 6.5  | 240.5 | 30   |
| S3  | H3         | Senegal.M    | 24.7 | 6.7  | 237.7 | 29.5 |
| S4  | H4         | Senegal.M    | 23.2 | 6    | 200.3 | 30.5 |
| S5  | H5         | Senegal.M    | 35.5 | 6.5  | 215.1 | 28.7 |
| S6  | H6         | Senegal.M    | 27.5 | 7.7  | 329.2 | 33.9 |
| S7  | H7         | Senegal.M    | 25.6 | 6.7  | 251.1 | 30.4 |
| S8  | H8         | Senegal.M    | 25.2 | 6.6  | 233.2 | 30.1 |
| S9  | H9         | Senegal.M    | 24.3 | 6.8  | 263.3 | 29.2 |
| S10 | H10        | Senegal.M    | 24.5 | 6.7  | 238.4 | 29.5 |
| S31 | 1B         | Senegal.M    | 21.6 | 5.6  | 144.1 | 26.2 |
| S11 | 1C         | Senegal.M    | 24   | 6.7  | 217.2 | 29   |
| S21 | 2B         | Senegal.M    | 24.4 | 6.8  | 259.5 | 29.7 |
| S22 | 2C         | Senegal.M    | 25.2 | 7    | 287.3 | 30.6 |
| S48 | 3B         | Senegal.M    | 22.5 | 6    | 197.7 | 27.1 |
| S49 | 3C         | Senegal.M    | 22.5 | 6    | 165.7 | 17   |
| S50 | 4C         | Senegal.M    | 22.2 | 5.8  | 152.5 | 26.1 |
| S41 | 4C-res     | Senegal.M    | 23.1 | 6    | 175   | 27.4 |
| S42 | 5C         | Senegal.M    | 21.3 | 6    | 175.3 | 27.7 |
| S43 | 9C         | Senegal.M    | 23   | 6.3  | 186.6 | 28.2 |
| G6  | 10E        | Guinea.M     | 15   | 5    | 48.8  | 18   |
| G7  | 6E         | Guinea.M     | 14.5 | 5    | 50.6  | 18   |
| G8  | 6F         | Guinea.M     | 14   | 5    | 38.9  | 17   |
| G9  | 7E         | Guinea.M     | 15   | 4.5  | 62.5  | 20   |
| G10 | 8E         | Guinea.M     | 15   | 6    | 46.4  | 17   |
| G23 | G23        | Guinea.M     | 14   | 4.5  | 36    | 19.5 |
| G25 | G25        | Guinea.M     | 14.5 | 5    | 55.4  | 17   |
| G26 | G26        | Guinea.M     | 15.5 | 5    | 45.6  | 19   |
| G27 | G27        | Guinea.M     | 14.5 | 5.5  | 58.1  | 17   |
| G28 | G28        | Guinea.M     | 14.5 | 5    | 45.6  | 20   |
| G29 | G29        | Guinea.M     | 14.5 | 5    | 58.9  | 18.5 |
| G30 | G30        | Guinea.M     | 15   | 5    | 55.4  | 18   |
| G33 | G33        | Guinea.M     | 15   | 5.5  | 56.6  | 18   |
| G34 | G34        | Guinea.M     | 14.5 | 5    | 54.7  | 17   |
| G35 | G35        | Guinea.M     | 14   | 4.5  | 40.5  | 17   |
| G37 | G37        | Guinea.M     | 13.5 | 5    | 53    | 19   |
| G38 | G38        | Guinea.M     | 13.5 | 5    | 50.4  | 18   |
| G39 | G39        | Guinea.M     | 15   | 5    | 39.6  | 6.5  |
| G40 | G40        | Guinea.M     | 13   | 4.5  | 45    | 17   |
| N1  | 3G         | Nigeria.M    | 12.2 | 10   | 30    | 14.5 |
| N2  | 3H         | Nigeria.M    | 13.8 | 10.6 | 28    | 16.2 |
| N3  | 8F         | Nigeria.M    | 13.1 | 10.2 | 25    | 15.3 |
| N4  | 1H         | Nigeria.M    | 20.5 | 14   | 40    | 24.7 |
| N5  | 10F        | Nigeria.M    | 13.6 | 9    | 38    | 16.1 |
| N6  | 10G        | Nigeria.M    | 13.2 | 8.7  | 35    | 15.4 |
| N7  | 10G.res    | Nigeria.M    | 13.3 | 8.8  | 35    | 16.6 |
| N8  | 11F        | Nigeria.M    | 20.3 | 14.1 | 40    | 24.3 |
| N9  | 11G        | Nigeria.M    | 13.3 | 8.8  | 35    | 15.8 |
| N10 | 12F        | Nigeria.M    | 12.7 | 8.3  | 30    | 14.8 |
| N11 | 12G        | Nigeria.M    | 13.3 | 8.6  | 35    | 15.5 |
| N12 | 1G         | Nigeria.M    | 13.6 | 9.1  | 45    | 16.3 |
| N13 | 2G         | Nigeria.M    | 14.3 | 9.3  | 25    | 17   |
| N14 | 2H         | Nigeria.M    | 12.2 | 8    | 25    | 14   |
| N15 | 4G         | Nigeria.M    | 14.2 | 12.2 | 35    | 16.8 |
| N16 | 5G         | Nigeria.M    | 13.9 | 9.2  | 80    | 16.3 |
| N17 | 7G         | Nigeria.M    | 16.9 | 8.5  | 55    | 20.3 |
| N18 | 8G         | Nigeria.M    | 13.9 | 9    | 30    | 16.3 |
| N19 | 9F         | Nigeria.M    | 12.7 | 9    | 35    | 15   |
| N20 | 9G         | Nigeria.M    | 13.1 | 9.9  | 40    | 15.5 |
| T1  | 10A        | Togo.M       | 21.5 | 13.5 | 142   | 26   |
| T2  | 10B        | Togo.M       | 20   | 11   | 129   | 24.5 |
| T3  | 11A        | Togo.M       | 17.5 | 13.5 | 73    | 20   |
| T4  | 11B        | Togo.M       | 20   | 12.5 | 123   | 24.5 |
| T5  | 12A        | Togo.M       | 19   | 13.5 | 104   | 22.5 |
| T6  | 12B        | Togo.M       | 21   | 13   | 141   | 25.5 |
| T7  | 3A         | Togo.M       | 20   | 12.5 | 117   | 24   |
| T8  | 4B         | Togo.M       | 20.5 | 13.5 | 137   | 25   |
| T9  | 5A         | Togo.M       | 21   | 13   | 147   | 25.5 |
| T10 | 5B         | Togo.M       | 22   | 14   | 171   | 26.5 |
| T11 | 6A         | Togo.M       | 22   | 14   | 153   | 21.5 |
| T12 | 6B         | Togo.M       | 18   | 11.5 | 91    | 26   |
| T13 | 8A         | Togo.M       | 21   | 13.5 | 143   | 25   |
| T14 | 8B         | Togo.M       | 21   | 13.5 | 129   | 24   |
| T15 | 8Bres      | Togo.M       | 21   | 13.5 | 69    | 20   |
| T26 | T26        | Togo.M       | 21.5 | 14.5 | 137   | 26   |
| T37 | T37        | Togo.M       | 20   | 11.5 | 71    | 20.5 |
| T38 | T38        | Togo.M       | 17   | 12.5 | 95    | 22.5 |
| T39 | T39        | Togo.M       | 19   | 13.5 | 118   | 23.5 |
| T45 | T45        | Togo.M       | 20   | 13   | 97    | 22.5 |



### PCA Plots of morphological data for *S. maderensis*

```df1 <- dataM[c(4,5,6)]
MYDATA <- autoplot(prcomp(df1))

autoplot(prcomp(df1), data = dataM, colour = 'Taxon')

autoplot(prcomp(df1), data = dataM, colour = 'Taxon', label=TRUE, shape = FALSE, label.size = 3, legend=F)

```
![plot1](https://github.com/evelyn-takyi/Sardinella-project/blob/master/maderensis_plot1.png)

![plot2](https://github.com/evelyn-takyi/Sardinella-project/blob/master/maderensis_plot2.png)


# Genetic Analysis
## This analysis was performed combining both reads of *s. aurita* and *S.maderensis*
### AIM: To show whether there is clear differentiation between *s.aurita* and *s.maderensis*

### Procedure

The ddocent pipeline, a protocol developed by Jon Puritz http://www.ddocent.com was used to assemble the references, map the reads and call SNPs combining all the sequences from *s.aurita* and *s.maderensis*

### SNP filtering
##### 1. this filters variants that have been successfully genotyped in 50% of individuals with  a minimum quality score of 20, and a minor allele count of 3
```
 vcftools --vcf TotalRawSNPs.vcf --max-missing 0.5 --mac 3 --minQ 20 --recode --recode-INFO-all --out raw.g5mac3
```
After filtering, kept 142840 out of a possible 985007 Sites

##### 2. filters loci with a minimum depth(--minDP) less than  3 and minor allele frequency(--maf) less than 1%
```
vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --maf 0.01 --recode --recode-INFO-all --out raw.g5mac3dp3
```
After filtering, kept 12807 out of a possible 15290 Sites

##### 3. Using the script (filter_missing_ind.sh)by Jon Puritz (https://github.com/jpuritz/dDocent/blob/master/scripts/filter_missing_ind.sh) to filter individuals with 90% missing data
```
./filter_missing_ind.sh raw.g5mac3dp3.recode.vcf DP3g95maf05
After filtering, kept 236 out of 236 Individuals
Outputting Individual Missingness
After filtering, kept 12807 out of a possible 12807 Sites

                                Histogram of % missing data per individual                                       

  10 +-+---------+------------+**---------+-----------+------------+-----------+-----------+------------+---------+-+   
     +           +            +**         +           +            +           +           +            +           +   
     |                         **                             'totalmissing' using (bin($1,binwidth)):(1.0) ******* |   
   9 +-+                       **                                                                                 +-+   
     |                         **                                                                                   |   
   8 +-+                   **  **                                                    **                           +-+   
     |                     **  **                                                    **                             |   
     |                     **  **                                                    **                             |   
   7 +-+                   **  **                                                    **                           +-+   
     |                     **  **                                                    **                             |   
   6 +-+                   **  *****                    **          **               ** **                        +-+   
     |                     **  *** *                    **          **               ** **                          |   
     |                     **  *** *                    **          **               ** **                          |   
   5 +-+                ** **  *** * **                 *******  *****  **    **     ** ** ***                    +-+   
     |                  ** **  *** * **                 **** **  * ***  **    **     ** ** * *                      |   
   4 +-+          ***   ** **  *** ****   ************  **** **  * ***  **    **     ** ** * *                    +-+   
     |            * *   ** **  *** ****   *** **** ***  **** **  * ***  **    **     ** ** * *                      |   
     |            * *   ** **  *** ****   *** **** ***  **** **  * ***  **    **     ** ** * *                      |   
   3 +-+          * * ************ ****** *** **** ***  **** ***** ***  ************ ******* *       ** **        +-+   
     |            * * * ***** **** **** * *** **** ***  **** ***** ***  **** ***** * ** **** *       ** **          |   
   2 +-+          * * * ***** **** **** ***** **** ********* ***** ********* ***** **** **** ****   ********      +-+   
     |            * * * ***** **** **** ***** **** **** **** ***** **** **** ***** **** **** ** *   *** ****        |   
     +           +* * * ***** **** **** ***** **** **** **** ***** **** **** ***** **** **** ** *   *** ****        +   
   1 +-+---------+***-***************************************************************************---********------+-+   
    0.1         0.2          0.3         0.4         0.5          0.6         0.7         0.8          0.9          1   
                                              % of missing data                                                   

The 85% cutoff would be 0.777309
Would you like to set a different cutoff, yes or no
no

After filtering, kept 204 out of 236 Individuals
Outputting VCF file...
After filtering, kept 12807 out of a possible 12807 Sites
```
##### 4. Using the script pop_missing_filter.sh by Jon Puritz(https://github.com/jpuritz/dDocent/blob/master/scripts/pop_missing_filter.sh) to filter loci with 75% missing data from each population
```
./pop_missing_filter.sh DP3g95maf05.recode.vcf popmap 0.75 1 TRSdp5MIp25
After filtering, kept 1344 out of a possible 12807 Sites
```
##### 5. Next code split the variant calls into SNP and indel genotypes
```
vcfallelicprimitives -k -g TRSdp5MIp25.recode.vcf | sed 's:\.|\.:\.\/\.:g' >TRSdp5MIp25g9.prim
```
##### 6. Next code removes the indels and retain only variants containing SNPs
```
vcftools --vcf TRSdp5MIp25g9.prim --recode-INFO-all --recode --out SNP.DP3g95maf05d --remove-indels
After filtering, kept 1789 out of a possible 1911 Sites
```
##### 7. Variants with maximum allele of two is retained
```
vcftools --vcf SNP.DP3g95maf05d.recode.vcf --recode-INFO-all --max-alleles 2 --out SNP.5a --recode
```
After filtering, kept 1585 out of a possible 1789 Sites

##### 8.  I run the program bayescan to detect neutral and outlier loci
##### First, the vcf file containing the SNPs is converted into a bayescan output format using the program PGDSpider
##### second, Run: BayeScan2.1_linux64bits SNPAM.BS -nbp 30 -thin 20
##### Using the script VCFtoOutlierOnly.sh (https://github.com/jpuritz/dDocent/blob/master/scripts/untested/VCFtoOutlierOnly.sh)  to output the neutral and outlier loci
```
./VCFtoOutlierOnly.sh SNPAM.recode.vcf SNPA_fst.txt  0.1 SNPAM
—SNPAM.outlieronly
After filtering, kept 204 out of 204 Individuals
After filtering, kept 382 out of a possible 1585 Sites
**Outputted 382 outliers**

--vcf SNPAM.recode.vcf
--out SNPAM.neutralonly
After filtering, kept 204 out of 204 Individuals
After filtering, kept 1203 out of a possible 1585 Sites
 **kept 1203 Sites**
```
##### 9. This code prints the number of neutral loci
```
mawk '!/#/' SNPAM.neutralonly.recode.vcf | wc -l
```
 **1203 neutral loci FINAL FILTERED SNPS**

## Analysis of the neutral SNPs to differentiate between *S.aurita* and *S.maderensis*

##### Packages loaded in R
```
library(vcfR)
library(readr)
library(ggplot2)
library(reshape2)
library(pegas)
library(ade4)
library(adegenet)
library(diveRsity)
library(genetics)
library(hierfstat)
library(iterators)
library(poppr)
library(readr)
library("mmod")
library(readr)
library(ade4)
library(genetics)
library(hierfstat)
library(poppr)
library(adegenet)
library(pegas)
library(StAMPP)
library(ggplot2)
library(ape)
library(ggmap)
library("devtools")
```
#### This code reads the SNP VCF file into R and converts it into a genlight  object.
```{r}
AU1 <- read.vcfR("~/Desktop/FINALANALYSIS/SNPAM.neutralonly.recode.vcf")
aa.genlight <- vcfR2genlight(AU1, n.cores=1)

locNames(aa.genlight) <- paste(AU1@fix[,1],AU1@fix[,2],sep="_")
pop(aa.genlight)<-substr(indNames(aa.genlight),1,3)
aa.genlight@position
as.matrix(aa.genlight)[1:16,1:10]

indNames(aa.genlight)
popNames(aa.genlight)
```
##### This code converts the file into a genind object
```
genindB <- vcfR2genind(AU1)
strata<- read.table("~/Desktop/FINALANALYSIS/popmap.txt", header=TRUE)
strata_df <- data.frame(strata)
strata(genindB) <- strata_df
setPop(genindB) <- ~Population
```
##### Determine the overall FST and FIS values
```
fstat(genindB)
wc(genindB)
```          
$FST
[1] 0.1279823

$FIS
[1] -0.2260168

##### construct a Neighbor joining tree
```
bov_pop <- genind2genpop(genindB)
set.seed(20150428)
pop_tree <- aboot(bov_pop, sample = 1000,
                           cutoff = 50)
```                           
![plot](https://github.com/evelyn-takyi/Sardinella-project/blob/master/All%20plot/jessy2.png)

##### PCA plot showing difference between s.aurita and s.maderensis
```
rubi.pca <- glPca(aa.genlight, nf = 3)
barplot(rubi.pca$eig/sum(rubi.pca$eig), col = heat.colors(50), main="PCA Eigenvalues")
title(ylab="Percent of variance\nexplained", line = 2)
title(xlab="Eigenvalues", line = 1)

rubi.pca.scores <- as.data.frame(rubi.pca$scores)
rubi.pca.scores$pop <- pop(aa.genlight)
```

![PCA plot](https://github.com/evelyn-takyi/Sardinella-project/blob/master/jessy1.png)



# Analysis for *s.aurita* species only

## Denovo Assembly of references

a. I run the script ReferenceOpt.sh(https://github.com/jpuritz/dDocent/blob/master/scripts/ReferenceOpt.sh)  to pick the cut off value for similarity threshold for clustering unique sequences together.
```
bash ReferenceOpt.sh 4 4 2 2 PE 20
```
[evelyn-takyi@bluewaves aurita_r]$ cat kopt.data 

| 4 | 2 | 0.8  | 23901   |
|---|---|------|---------|
| 4 | 2 | 0.8  | 2 23952 |
| 4 | 2 | 0.84 | 23986   |
| 4 | 2 | 0.86 | 24026   |
| 4 | 2 | 0.88 | 24047   |
| 4 | 2 | 0.9  | 24071   |
| 4 | 2 | 0.92 | 24171   |
| 4 | 2 | 0.94 | 24276   |
| 4 | 2 | 0.96 | 24392   |
| 4 | 2 | 0.98 | 24503   |

![plot1](https://github.com/evelyn-takyi/Sardinella-project/blob/master/RefoptM.png)

The similarity threshold of 90% is appropriate cut off value.

b. The script RefMapOpt.sh (https://github.com/jpuritz/dDocent/blob/master/scripts/RefMapOpt.sh) was run to pick K1(unique sequences within a coverage that  appear in most of the individuals for the assembly), K2(cut off value of the number of individuals a unique sequence should atleast appear in) values which show properly paired mapping and good coverage and less number of  mismatched reads.

```
bash RefMapOpt.sh 2 5 2 5 0.9 20 PE
```
Mapping results

| Cov     | Non0Cov | Contigs | MeanContigsMapped | K1 | K2 | SUM Mapped | SUM Properly | Mean Mapped | Mean Properly | MisMatched |
|---------|---------|---------|-------------------|----|----|------------|--------------|-------------|---------------|------------|
| 5.1378  | 7.52733 | 258610  | 172140            | 2  | 2  | 18601681   | 13233039     | 1.33E+06    | 945217        | 91745.6    |
| 7.22865 | 9.45369 | 130928  | 95859.9           | 2  | 3  | 13250174   | 8658665      | 946441      | 618476        | 66231.3    |
| 10.0434 | 12.6844 | 74720   | 56594.2           | 2  | 4  | 10506338   | 6621964      | 750453      | 472997        | 55011      |
| 14.6825 | 18.116  | 42702   | 33203.8           | 2  | 5  | 8777794    | 5211280      | 626985      | 372234        | 41542.9    |
| 9.67957 | 12.8533 | 82324   | 60523.7           | 3  | 2  | 11156193   | 7219651      | 796871      | 515689        | 60601.2    |
| 18.1624 | 22.0825 | 30121   | 23726.6           | 3  | 3  | 7659226    | 4561326      | 547088      | 325809        | 38603.5    |
| 36.3081 | 42.6345 | 13061   | 10699.9           | 3  | 4  | 6639589    | 3534219      | 474256      | 252444        | 33968.8    |
| 64.7652 | 73.2299 | 6317    | 5359.21           | 3  | 5  | 5728612    | 3008507      | 409187      | 214893        | 26955.6    |
| 23.3165 | 28.935  | 22878   | 17971.8           | 4  | 2  | 7468413    | 4711947      | 533458      | 336568        | 40834.8    |
| 61.7502 | 69.8452 | 6976    | 5897.79           | 4  | 3  | 6031633    | 3205327      | 430831      | 228952        | 26372.8    |
| 116.116 | 127.164 | 3394    | 2974.07           | 4  | 4  | 5518973    | 2795911      | 394212      | 199708        | 24969.1    |
| 154.138 | 165.389 | 2188    | 1946.71           | 4  | 5  | 4723704    | 2600495      | 337407      | 185750        | 16348.9    |
| 59.8067 | 70.5072 | 7680    | 6360.57           | 5  | 2  | 6431255    | 3533321      | 459375      | 252380        | 38750.4    |
| 120.921 | 129.114 | 2838    | 2506.57           | 5  | 3  | 4806145    | 2791040      | 343296      | 199360        | 16342.9    |
| 173.338 | 182.504 | 1806    | 1615.43           | 5  | 4  | 4385096    | 2525316      | 313221      | 180380        | 17047.4    |
| 217.311 | 227.027 | 1346    | 1214.14           | 5  | 5  | 4098051    | 2336437      | 292718      | 166888        | 15973.4    |

c. The script  remake_reference.sh was used to remake the reference with the cut off value of K1=3, K2=2, and a c(similarity threshold) value of 0.9

```
bash remake_reference.sh 3 2 0.90 PE 20
mawk '/>/' reference.fasta | wc -l
```
**82324** Contigs were assembled for *S.aurita*

## Blast sequences of *s.aurita* published on ncbi against the denovo reference assembly

```
Database: Aurita
           82,324 sequences; 23,834,470 total letters
```

| Query                                                                                                                          | Length of query | Sequences producing significant alignments: | Bits Score | E-value   |
|--------------------------------------------------------------------------------------------------------------------------------|-----------------|---------------------------------------------|------------|-----------|
|                                                                                                                                |                 |                                             |            |           |
|                                                                                                                                |                 |                                             |            |           |
| DQ197990.1 Sardinella aurita cytochrome b (cytb) gene, complete cds;mitochondrial                                              | 1141            | dDocent_Contig_411                          | 483        | 9.00E-136 |
|                                                                                                                                |                 | dDocent_Contig_1873                         | 425        | 2.00E-118 |
|                                                                                                                                |                 | dDocent_Contig_2480                         | 418        | 3.00E-116 |
|                                                                                                                                |                 | dDocent_Contig_1242                         | 368        | 3.00E-101 |
|                                                                                                                                |                 | dDocent_Contig_16133                        | 268        | 3.00E-71  |
|                                                                                                                                |                 | dDocent_Contig_77003                        | 263        | 1.00E-69  |
| KJ655528.1 Sardinella aurita clone Sara8 microsatellite sequence                                                               | 595             | dDocent_Contig_54832                        | 204        | 4.00E-52  |
|                                                                                                                                |                 | dDocent_Contig_21255                        | 126        | 9.00E-29  |
|                                                                                                                                |                 |                                             |            |           |
| Sardinella aurita voucher JFBM 48669-JE1131 recombination activating protein 1 gene, partial cds                               | 1471            | dDocent_Contig_72123                        | 268        | 4.00E-71  |
|                                                                                                                                |                 |                                             |            |           |
|                                                                                                                                |                 |                                             |            |           |
| MG958416.1 Sardinella aurita voucher MNHN-IC 2017-0508-BPS-1383 recombination activating protein 1 gene, partial cds           | 1510            | dDocent_Contig_72123                        | 274        | 8.00E-73  |
|                                                                                                                                |                 |                                             |            |           |
| MG958277.1 Sardinella aurita voucher JFBM 48669-JE1131 recombination activating protein 2 gene, partial cds                    | 1122            | dDocent_Contig_2087                         | 342        | 2.00E-93  |
|                                                                                                                                |                 |                                             |            |           |
| Sardinella aurita voucher JFBM 48669-JE1131 cytochrome b gene, partial cds; mitochondrial                                      | 1142            | dDocent_Contig_411                          | 484        | 3.00E-136 |
|                                                                                                                                |                 | dDocent_Contig_1873                         | 456        | 7.00E-122 |
|                                                                                                                                |                 | dDocent_Contig_2480                         | 401        | 3.00E-111 |
|                                                                                                                                |                 | dDocent_Contig_1242                         | 372        | 2.00E-102 |
|                                                                                                                                |                 | dDocent_Contig_16133                        | 263        | 1.00E-69  |
|                                                                                                                                |                 | dDocent_Contig_77003                        | 257        | 6.00E-68  |
|                                                                                                                                |                 | dDocent_Contig_43148                        | 52.8       | 3.00E-06  |
| MG958159.1 Sardinella aurita voucher MNHN-IC 2017-0508-BPS-1383 zic family member 1 gene, partial cds                          | 812             | dDocent_Contig_44444                        | 357        | 4.00E-98  |
|                                                                                                                                |                 |                                             |            |           |
| Sardinella aurita isolate SAUR5 tRNA-Glu (trnE) gene,partial sequence; and cytochrome b (cytb) gene, partial cds;mitochondrial | 414             | dDocent_Contig_411                          | 466        | 3.00E-131 |
|                                                                                                                                |                 | dDocent_Contig_1242                         | 351        | 9.00E-97  |
|                              


### The dDocent pipeline was used to map all the reads to the above reference assembled and SNP calling

### Filtering of raw SNP calls using vcftools and bash scripts(http://ddocent.com/filtering/)

```
 vcftools --vcf TotalRawSNPs.vcf --max-missing 0.5 --mac 3 --minQ 20 --recode --recode-INFO-all --out raw.g5mac3
After filtering, kept 115 oOutputting VCF file...
After filtering, kept 341983 out of a possible 1256073 Sites
```
```
./filter_missing_ind.sh raw.g5mac3.recode.vcf DP3g95maf05

        Histogram of % missing data per individual                                       

  7 +-+---**---+----------+-----------+----------+----------+----------+----------+-----------+----------+--------+-+   
    +     **   +          +           +          +          +          +          +           +          +          +   
    |     **                                                  'totalmissing' using (bin($1,binwidth)):(1.0) ******* |   
    |     **                                                                                                        |   
  6 +-+   ***                                                                                                     +-+   
    |     ***                                                                                                       |   
    |     ***                                                                                                       |   
    |     ***                                                                                                       |   
  5 +-+   ***                                                                                                     +-+   
    |     ***                                                                                                       |   
    |     ***                                                                                                       |   
  4 +**** ******* ***                                                                                             +-+   
    |**** ******* * *                                                                                               |   
    |**** ******* * *                                                                                               |   
    |**** ******* * *                                                                                               |   
  3 +**** ******* * *               ***                                                                           +-+   
    |**** ******* * *               * *                                                                             |   
    |**** ******* * *               * *                                                                             |   
    |**** ******* * *               * *                                                                             |   
  2 +************** ****    ****  *** *  *****    *** ****      ***               ****** ** ***             ***   +-+   
    |**** ******* * ** *    *  *  * * *  * ***    * * *  *      * *               * **** ** * *             * *     |   
    |**** ******* * ** *    *  *  * * *  * ***    * * *  *      * *               * **** ** * *             * *     |   
    +**** ******* * ** *  + *  *  * * *  * ***   +* * *  *  +   * *    +          * **** ** * *          +  * *     +   
  1 +*******************--+-****--*****--*****---+***-****--+---***----+----------******-**-***----------+--***---+-
  0         0.1        0.2         0.3        0.4        0.5        0.6        0.7         0.8     0.9         1   
                                              % of missing data*                                                   


After filtering, kept 100 out of 115 Individuals

After filtering, kept 341983 out of a possible 341983 Sites

```

```
vcftools --vcf DP3g95maf05.recode.vcf --minDP 5 --maf 0.01 --recode --recode-INFO-all --out raw.g5mac3dp3
```
After filtering, kept 276130 out of a possible 341983 Sites

```
./pop_missing_filter.sh raw.g5mac3dp3.recode.vcf popmapF 0.9 1 TRSdp5MIp25
```

After filtering, kept 58019 out of a possible 276130 Sites

```
./dDocent_filters TRSdp5MIp25.recode.vcf  TRSdp5MIp25g9dd

Number of sites filtered based on allele balance at heterozygous loci, locus quality, and mapping quality / Depth
37787 of 58019

Are reads expected to overlap?  In other words, is fragment size less than 2X the read length?  Enter yes or no.
no
Number of additional sites filtered based on overlapping forward and reverse reads
11084 of 20232

Is this from a mixture of SE and PE libraries? Enter yes or no.
no
Number of additional sites filtered based on properly paired status
324 of 9148

Number of sites filtered based on high depth and lower than 2*DEPTH quality score
4206 of 8824



                                             Histogram of mean depth per site                                           

  400 +-+--+----+-----+----+----+----+-----+----+----+----+-----+----+----+----+-----+----+----+----+-----+----+--+-+   
      +    +    +     +    +    +    +     +    +    +    +     +    +    +    +     +    +    +    +     +    +    +   
      ****                                                'meandepthpersite' using (bin($1,binwidth)):(1.0) ******* |   
  350 *-+*                                                                                                        +-+   
      *  *                                                                                                          |   
      *  *                                                                                                          |   
  300 *-+*                                                                                                        +-+   
      *  *                                                                                                          |   
      *  ***                                                                                                        |   
  250 *-+* *                                                                                                      +-+   
      *  * *                                                                                                        |   
  200 *-+* *                                                                                                      +-+   
      *  * ****                                                                                                     |   
      *  * *  *                                                                                                     |   
  150 *-+* *  *                                                                                                   +-+   
      *  * *  ******    ****                                                                                        |   
      *  * *  * *  ******  *                                                                                        |   
  100 *-+* *  * *  *  * *  ******                                                                                 +-+   
      *  * *  * *  *  * *  *  * *                                                                                   |   
      *  * *  * *  *  * *  *  * **************                                                                      |   
   50 *-+* *  * *  *  * *  *  * *  * *  *  * *******    *********                                                 +-+   
      *  * *  * *  *  * *  *  * *  * *  *  * *  *  ****** *  *  *      ****       ****                              |   
      *  * *  * *  *  * *  *  * *  * *  *  * *  *  * *  * *  *  ********  *********  ********************************   
    0 ***************************************************************************************************************   
      10   12   14    16   18   20   22    24   26   28   30    32   34   36   38    40   42   44   46    48   50   52  
                                                        Mean Depth                                                      

If distrubtion looks normal, a 1.645 sigma cutoff (~90% of the data) would be 4927.04015
The 95% cutoff would be 48
Would you like to use a different maximum mean depth cutoff than 48, yes or no
no
Number of sites filtered based on maximum mean depth
564 of 8824

Number of sites filtered based on within locus depth mismatch
96 of 8260

Total number of sites filtered
49855 of 58019

Remaining sites
8164
```

```
vcftools --vcf TRSdp5MIp25g9dd.FIL.recode.vcf --maf 0.05 --recode --recode-INFO-all --out DP3g95maf05d
```
After filtering, kept 6400 out of a possible 8164 Sites

```
filter_hwe_by_pop.pl -v DP3g95maf05d.recode.vcf -p popmap -c 0.5 -o SNP.DP3gdp5MIp25g9HWE
Processing population: BA (17 inds)
Processing population: GA (20 inds)
Processing population: GUI (20 inds)
Processing population: MA (19 inds)
Processing population: SA (18 inds)
Processing population: TA (20 inds)
Outputting results of HWE test for filtered loci to 'filtered.hwe'
Kept 6116 of a possible 6400 loci (filtered 284 loci)
```

```
rad_haplotyper.pl -v SNP.DP3gdp5MIp25g9HWE.recode.vcf  -x 10 -mp 2 -r reference.fasta
```
less stats.out

```(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ cp stats.out stats.out.HF
(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ mawk '/Missi/' stats.out.HF | mawk '$9 > 30' > HF.missing
(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ mawk '/para/' stats.out.HF > HF.para
(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ ./remove.bad.hap.loci.sh HF.loci.tofilter  SNP.DP3gdp5MIp25g9HWE.recode.vcf  
(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ mawk '!/#/' SNP.DP3gdp5MIp25g9HWE.filtered.vcf | wc -l
2654 SNP loci
```

```vcfallelicprimitives -k -g SNP.DP3gdp5MIp25g9HWE.filtered.vcf| sed 's:\.|\.:\.\/\.:g' > TRSdp5MIp25g9.prim
(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ vcftools --vcf TRSdp5MIp25g9.prim --recode-INFO-all --recode --out SNP.DP3g95maf05d --remove-indels
```
After filtering, kept 2847 out of a possible 3202 Sites

```
vcftools --vcf SNP.DP3g95maf05d.recode.vcf  --recode-INFO-all --max-alleles 2 --out SNP.2a --recode
```
After filtering, kept 2776 out of a possible 2847 Sites

```
vcftools --vcf SNP.2a.recode.vcf --recode-INFO-all --max-missing 0.5 --out SNP.dp5a --recode
```
After filtering, kept 1393 out of a possible 2776 Sites

```
mawk '!/#/' SNP.dp5a.recode.vcf | wc -l
```
1393 SNPs loci

##### PGDSpider was used to convert SNPs to BayeScan input file.
```
BayeScan2.1_linux64bits SNP1.BS -thin 100 -nbp 30
```
##### A VCF file of SNPs of outliers loci and neutral loci was generated with a false discovery rate of 10% using VCFtoOutlierOnly.sh


```
./VCFtoOutlierOnly.sh SNP.dp5a.recode.vcf  SNPDP5_fst.txt  0.1 SNPAUdp  

	--out SNPAUdp.outlieronly
After filtering, kept 11 outliers out of a possible 1393 Sites

	--out SNPAUdp.neutralonly

After filtering, kept 100 out of 100 Individuals
After filtering, kept 1382 neutral loci out of a possible 1393 Sites
```

### Analysis of neutral SNPs using

Basic statistics from running populations program in STACKS package in bluewaves
```
code:
populations -V SNP.dp5a.recode.vcf  -O /data3/marine_diseases_lab/evelyn/eveproject/evelyn/gridftp.tamucc.edu/genomics/20180312-Takyi-Sardinella/Aurita_Final2/ -M popmap --fstats --fst_correction p_value  --bootstrap  --kernel_smooth --genepop --structure
```

| Population | Private Alleles | Sites | Polymorphic Sites | % Polymorphic Loci | Obs Het | Exp Het | Nucleotide diversity | Fis     |
|------------|-----------------|-------|-------------------|--------------------|---------|---------|----------------------|---------|
| Benin      | 0               | 1383  | 1330              | 96.1678            | 0.6296  | 0.3823  | 0.4013               | -0.4602 |
| Ghana      | 0               | 1383  | 1333              | 96.3847            | 0.6104  | 0.3744  | 0.3909               | -0.4462 |
| Guinea     | 0               | 1383  | 967               | 69.9205            | 0.5388  | 0.308   | 0.4414               | -0.173  |
| Mauritania | 0               | 1383  | 1308              | 94.577             | 0.6269  | 0.3759  | 0.3902               | -0.4808 |
| Senegal    | 0               | 1383  | 1355              | 97.9754            | 0.6317  | 0.3895  | 0.4045               | -0.467  |
| Togo       | 0               | 1383  | 1359              | 98.2646            | 0.6246  | 0.3876  | 0.401                | -0.4631 |



#### The overall FST and FIS values were calculated using the diffcalc function in the genepop package in R
```test_result <- diffCalc(infile = "SNPAUdp.neutralonly.recode.gen", outfile = "myresults",fst = TRUE, pairwise = TRUE, bs_locus = TRUE, bs_pairwise = TRUE, boots = 1000, para = TRUE)
```
overall FST  = 0.0080	 CI (0.0019-0.0157)
overall FIS  = -0.6061 CI	()-0.6557 - -0.5746)

#### Pairwise FST values calculated using the program Genodive
 Values of Fst for all pairs of populations is significant at (p<0.05)

  |            | Benin  | Ghana | Guinea | Mauritania | Senegal |
 |------------|--------|-------|--------|------------|---------|
 | Benin      |        |       |        |            |         |
 | Ghana      | -0.002 |       |        |            |         |
 | Guinea     | 0.05   | 0.063 |        |            |         |
 | Mauritania | 0      | 0.001 | 0.072 |            |         |
 | Senegal    | 0.004  | 0.007 | 0.029  | 0.009      |         |
 | Togo       | 0.003  | 0.007 | 0.029  | 0.009      | -0.002  |

p-values for all pairs of populations
|            | Benin  | Ghana | Guinea | Mauritania | Senegal |
|------------|--------|-------|--------|------------|---------|
| Benin      |        |       |        |            |         |
| Ghana      | 0.823  |       |        |            |         |
| Guinea     | 0.00   | 0.00  |        |            |         |
| Mauritania | 0.53   | 0.13  | 0.00   |            |         |
| Senegal    | 0.07   | 0.022 | 0.005  | 0.007      |         |
| Togo       | 0.095  | 0.01  | 0.002  | 0.002      | 0.814   |



#### Neighbor joining tree was generated using the function aboot in the poppr package in R
```{r}
bov_pop <- genind2genpop(genindB)
set.seed(20150428)
pop_tree <- aboot(bov_pop, sample = 1000, cutoff = 50)
```
![Plot](https://github.com/evelyn-takyi/Sardinella-project/blob/master/All%20plot/jessy3.png)


#### PCA plot was generated using the package  genlight in R
```{r}
rubi.pca <- glPca(aa.genlight, nf = 3)
barplot(rubi.pca$eig/sum(rubi.pca$eig), col = heat.colors(50), main="PCA Eigenvalues")
title(ylab="Percent of variance\nexplained", line = 2)
title(xlab="Eigenvalues", line = 1)

rubi.pca.scores <- as.data.frame(rubi.pca$scores)
rubi.pca.scores$pop <- pop(aa.genlightA)

library(ggplot2)
set.seed(9)
p <- ggplot(rubi.pca.scores, aes(x=PC1, y=PC2, colour=pop))
p <- p + geom_point(size=2)
p <- p + stat_ellipse(level = 0.95, size = 1)
p <- p + geom_hline(yintercept = 0)
p <- p + geom_vline(xintercept = 0)
p <- p + theme_bw()
p
```
![plot2](https://github.com/evelyn-takyi/Sardinella-project/blob/master/All%20plot/jessy4.png)                       


#### Discriminant analysis of principal components (DAPC) was conducted in R using  the package Adegenet 1.3.0 to generate the posterior membership probability of each sample

```
set.seed(20160308) # Setting a seed for a consistent result
grp <- find.clusters(genindB, max.n.clust = 10, n.pca = 45, choose.n.clust = TRUE)
names(grp)
grp$grp
dapc1 <- dapc(genindB, grp$grp, n.pca = 10, n.da = 2)
scatter(dapc1) # plot of the group
```
Bayesian Criterion Information to determine the number of clusters
![Dapc_BIC](https://github.com/evelyn-takyi/Sardinella-project/blob/master/dapc_BIC.png)

#### structure-like plot using Posterior probability from dapc function

```
## dapcB.results <- as.data.frame(dapc1$posterior)
dapcB.results$pop <- pop(genindB)
dapcB.results$indNames <- rownames(dapcB.results)
library(reshape2)
dapcB.results <- melt(dapcB.results)
colnames(dapcB.results) <- c("Original_Pop","Sample","Assigned_Pop","Posterior_membership_probability")

p <- ggplot(dapcB.results, aes(x=Sample,y=Posterior_membership_probability, fill=Assigned_Pop))
p <- p + geom_bar(stat='identity')
p <- p + facet_grid(~Original_Pop, scales = "free")
p <- p + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 5))
p

calculate mean per population
grouped_data <- group_by(dapcB.results, Original_Pop, Assigned_Pop)
data_means <- summarise(grouped_data, mean=mean(Posterior_membership_probability))
# plot means for each original pop colored by assigned pop
pieB <- ggplot(data_means, aes(x=Original_Pop,y=mean, fill=Assigned_Pop))
pieB <- pieB + geom_bar(stat='identity') + cp
pieB <- pieB + facet_grid(~Original_Pop, scales = "free") + theme(axis.text=element_blank(), axis.ticks=element_blank(), panel.grid = element_blank(), strip.background = element_blank())
pieB


cp <- coord_polar(theta = "y")
cp$is_free <- function() TRUE
```
structure-like plot
![Plot3](https://github.com/evelyn-takyi/Sardinella-project/blob/master/All%20plot/jessy5.png)

#### Isolation by distance
```{r}
xy <- read.table("~/Desktop/AMresults/S_auritadismatrix.txt")
genindB@other$xy <- xy
genindB

titi <- genind2genpop(genindB)
genetic_distance <- dist.genpop(titi, method=2)
geographic_distance <- dist(titi$other$xy)
plot(geographic_distance, genetic_distance)
abline(lm(genetic_distance~geographic_distance), col="red",lty=2)
ibd <- mantel.randtest(genetic_distance,geographic_distance)
plot(ibd)
```
![plot4](https://github.com/evelyn-takyi/Sardinella-project/blob/master/All%20plot/jessy6.png)


## S. maderensis
### Denovo Assembly of references

a. I run the script ReferenceOpt.sh  to pick the cut off values of K1 and K2 and a similarity threshold.
```
bash ReferenceOpt.sh 4 4 2 2 PE 20
```
###### cat kopt.data 
 kopt.data

| 4 | 2 | 0.80 | 20285 |
|---|---|------|-------|
| 4 | 2 | 0.82 | 20323 |
| 4 | 2 | 0.84 | 20364 |
| 4 | 2 | 0.86 | 20383 |
| 4 | 2 | 0.88 | 20400 |
| 4 | 2 | 0.90 | 20422 |
| 4 | 2 | 0.92 | 20520 |
| 4 | 2 | 0.94 | 20623 |
| 4 | 2 | 0.96 | 20721 |
| 4 | 2 | 0.98 | 20845 |

![plot2](https://github.com/evelyn-takyi/Sardinella-project/blob/master/RefoptA.png)

The similarity threshold of 90% is observed to be the appropriate cut off value.

b. The script RefMapOpt.sh was run to pick K1, K2 values which were properly paired and good coverage, minimizing mismatched reads.

```
bash RefMapOpt.sh 2 5 2 5 0.9 20 PE
```
Kopt data

| 2 | 2 | 0.80 | 181457 |
|---|---|------|--------|
| 2 | 2 | 0.82 | 182072 |
| 2 | 2 | 0.84 | 182606 |
| 2 | 2 | 0.86 | 183057 |
| 2 | 2 | 0.88 | 183387 |
| 2 | 2 | 0.90 | 183620 |
| 2 | 2 | 0.92 | 185130 |
| 2 | 2 | 0.94 | 186665 |
| 2 | 2 | 0.96 | 188208 |
| 2 | 2 | 0.98 | 189824 |
| 2 | 3 | 0.80 | 60780  |
| 2 | 3 | 0.82 | 60919  |
| 2 | 3 | 0.84 | 61047  |
| 2 | 3 | 0.86 | 61153  |
| 2 | 3 | 0.88 | 61214  |
| 2 | 3 | 0.90 | 61264  |
| 2 | 3 | 0.92 | 61558  |
| 2 | 3 | 0.94 | 61894  |
| 2 | 3 | 0.96 | 62214  |
| 2 | 3 | 0.98 | 62552  |
| 2 | 4 | 0.80 | 25658  |
| 2 | 4 | 0.82 | 25698  |
| 2 | 4 | 0.84 | 25745  |
| 2 | 4 | 0.86 | 25777  |
| 2 | 4 | 0.88 | 25797  |
| 2 | 4 | 0.90 | 25816  |
| 2 | 4 | 0.92 | 25900  |
| 2 | 4 | 0.94 | 26015  |
| 2 | 4 | 0.96 | 26117  |
| 2 | 4 | 0.98 | 26221  |
| 2 | 5 | 0.80 | 11647  |
| 2 | 5 | 0.82 | 11666  |
| 2 | 5 | 0.84 | 11678  |
| 2 | 5 | 0.86 | 11690  |
| 2 | 5 | 0.88 | 11697  |
| 2 | 5 | 0.90 | 11704  |
| 2 | 5 | 0.92 | 11741  |
| 2 | 5 | 0.94 | 11783  |
| 2 | 5 | 0.96 | 11828  |
| 2 | 5 | 0.98 | 11868  |
Mapping results

|         |         |         |                   |    |    |            |              |             |
|---------|---------|---------|-------------------|----|----|------------|--------------|-------------|
| Cov     | Non0Cov | Contigs | MeanContigsMapped | K1 | K2 | SUM Mapped | SUM Properly | Mean Mapped |
| 7.53407 | 12.65   | 183620  | 114239            | 2  | 2  | 16600951   | 11344500     | 1.38E+06    |
| 14.1841 | 21.3783 | 61264   | 43034.2           | 2  | 3  | 10427872   | 6597936      | 868989      |
| 24.4562 | 35.1343 | 25816   | 19187.4           | 2  | 4  | 7576629    | 4590766      | 631386      |
| 16.8484 | 25.7664 | 50495   | 34515.1           | 3  | 2  | 10209312   | 6458448      | 850776      |
| 42.9016 | 60.4159 | 13624   | 10106.7           | 3  | 3  | 7014410    | 3968219      | 584534      |
| 85.9141 | 114.759 | 5118    | 3957.25           | 3  | 4  | 5277534    | 2940855      | 439794      |
| 39.1148 | 56.5291 | 16092   | 11408.6           | 4  | 2  | 7553693    | 4321675      | 629474      |
| 102.171 | 139.566 | 4451    | 3323.25           | 4  | 3  | 5458361    | 2912050      | 454863      |
| 179.742 | 236.671 | 1955    | 1527.83           | 4  | 4  | 4218912    | 2390190      | 351576   


c. The script  remake_reference.sh was used to remake the reference with the cut off value of K1=2, K2=3, and a c value of 0.9

```
bash remake_reference.sh 2 3 0.90 PE 20
mawk '/>/' reference.fasta | wc -l
```
**61264** Contigs were assembled for *S.maderensis*

## Blast sequences from ncbi to reference assembly
#### build a database of the reference assembly
```
makeblastdb -in reference.fasta -parse_seqids  -title "mad" -dbtype nucl
```
```
Building a new DB, current time: 02/13/2019 22:23:18
New DB name:   /net/fs03.cluster.com/data3/marine_diseases_lab/evelyn/eveproject/evelyn/gridftp.tamucc.edu/genomics/20180312-Takyi-Sardinella/maderensis_Final2/No_NM/reference/reference.fasta
New DB title:  mad
Sequence type: Nucleotide
Keep Linkouts: T
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 61264 sequences in 2.07483 seconds.
```
#### perform a nucleotide blast on references
```
blastn -query sequence.fasta  -db reference.fasta.3 -out results.txt2
```
```
Database: maderensis
           61,264 sequences; 17,686,769 total letters
Query= DQ197991.1 Sardinella maderensis cytochrome b (cytb) gene, complete
cds; mitochondrial
Length=1141
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value
dDocent_Contig_21512                                                  436     5e-122
dDocent_Contig_3189                                                   366     7e-101
dDocent_Contig_18897                                                  351     2e-96
dDocent_Contig_19547                                                  316     7e-86
dDocent_Contig_688                                                    233     8e-61

Query= MG958221.1 Sardinella maderensis voucher MNHN-IC 2013-0981-BPS-2657
cytochrome b gene, partial cds; mitochondrial
Length=1143
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value

dDocent_Contig_21512                                                  436     5e-122
dDocent_Contig_3189                                                   372     2e-102
dDocent_Contig_18897                                                  357     4e-98
dDocent_Contig_19547                                                  300     7e-81
dDocent_Contig_688                                                    259     1e-68

Query= KY176607.1 Sardinella maderensis voucher 981 cytochrome c oxidase
subunit I (COI) gene, partial cds; mitochondrial
Length=642
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value

dDocent_Contig_839                                                    527     2e-149
dDocent_Contig_262                                                    305     9e-83
dDocent_Contig_957                                                    265     1e-70
dDocent_Contig_10743                                                  233     4e-61

Query= KX062183.1 Sardinella maderensis isolate SMAD4 tRNA-Glu (trnE) gene,
partial sequence; and cytochrome b (cytb) gene, partial cds;
mitochondrial
Length=414
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value
dDocent_Contig_3189                                                   361     1e-99
dDocent_Contig_19547                                                  335     7e-92
dDocent_Contig_18897                                                  289     6e-78

Query= KP307745.1 Sardinella maderensis voucher TAU<ISR_:P.15439 16S
ribosomal RNA gene, partial sequence; mitochondrial
Length=586
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value
dDocent_Contig_8343                                                   327     2e-89
dDocent_Contig_1361                                                   148     1e-35

Query= NC_009587.1 Sardinella maderensis mitochondrion, complete genome
Length=16651
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value
dDocent_Contig_23265                                                  534     3e-150
dDocent_Contig_839                                                    532     1e-149
dDocent_Contig_2057                                                   520     8e-146
dDocent_Contig_985                                                    503     8e-141
dDocent_Contig_6519                                                   497     4e-139
dDocent_Contig_22457                                                  496     1e-138
dDocent_Contig_40235                                                  494     5e-138
dDocent_Contig_688                                                    468     3e-130
dDocent_Contig_1720                                                   464     4e-129
dDocent_Contig_19547                                                  460     5e-128
dDocent_Contig_2959                                                   459     2e-127
dDocent_Contig_3936                                                   457     6e-127
dDocent_Contig_8343                                                   453     8e-126
dDocent_Contig_21512                                                  448     4e-124
dDocent_Contig_1361                                                   409     2e-112
dDocent_Contig_262                                                    407     6e-112
dDocent_Contig_10236                                                  398     4e-109
dDocent_Contig_8419                                                   383     1e-104
dDocent_Contig_965                                                    381     4e-104
dDocent_Contig_3189                                                   377     5e-103
dDocent_Contig_1696                                                   377     5e-103
dDocent_Contig_18897                                                  357     6e-97
dDocent_Contig_15907                                                  357     6e-97
dDocent_Contig_434                                                    320     8e-86
dDocent_Contig_8029                                                   316     1e-84
dDocent_Contig_5320                                                   316     1e-84
dDocent_Contig_312                                                    315     4e-84
dDocent_Contig_52690                                                  292     2e-77
dDocent_Contig_46683                                                  285     3e-75
dDocent_Contig_54810                                                  279     1e-73
dDocent_Contig_15265                                                  279     1e-73
dDocent_Contig_3660                                                   279     1e-73
dDocent_Contig_957                                                    279     1e-73
dDocent_Contig_4870                                                   278     5e-73
dDocent_Contig_46861                                                  274     7e-72
dDocent_Contig_24625                                                  272     2e-71
dDocent_Contig_58                                                     272     2e-71
dDocent_Contig_2463                                                   268     3e-70
dDocent_Contig_10743                                                  233     1e-59
dDocent_Contig_901                                                    224     7e-57
dDocent_Contig_14455                                                  196     1e-48
dDocent_Contig_444                                                    193     2e-47

Query= AM911205.1 Sardinella maderensis mitochondrial partial 16S rRNA gene
Length=568
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value
dDocent_Contig_8343                                                   329     5e-90
dDocent_Contig_1361                                                   141     2e-33

Query= AM911175.1 Sardinella maderensis mitochondrial partial coi gene for cytochrome oxidase subunit I
Length=594
                                                                     E
Sequences producing significant alignments:                          (Bits)  Value
dDocent_Contig_839                                                    532     3e-151
dDocent_Contig_262                                                    289     8e-78
dDocent_Contig_957                                                    265     1e-70
dDocent_Contig_10743                                                  233     4e-61
```


### The dDocent pipeline was used to map all the reads to reference assembly and SNP calling

#### Filtering of raw SNP calls using vcftools and bash scripts
##### filtering codes

```
vcftools --vcf TotalRawSNPs.vcf --max-missing 0.5 --mac 3 --minQ 20 --recode --recode-INFO-all --out raw.g5mac3
```
After filtering, kept 142840 out of a possible 985007 Sites

```
/filter_missing_ind.sh raw.g5mac3.recode.vcf DP3g95maf05
```
After filtering, kept 142840 out of a possible 142840 Sites


```

                                       Histogram of % missing data per individual                                       

  6 +-+-----**-+----------+-----------+----------+----------+----------+----------+-----------+----------+--------+-+   
    +       ** +          +           +          +          +          +          +           +          +          +   
    |       **                                                'totalmissing' using (bin($1,binwidth)):(1.0) ******* |   
    |       **                                                                                                      |   
    |       **                                                                                                      |   
  5 +-+   ****                                                                       **                           +-+   
    |     ****                                                                       **                             |   
    |     ****                                                                       **                             |   
    |     ****                                                                       **                             |   
  4 +-+   *****      **                     ***                                      **                           +-+   
    |     *****      **                     * *                                      **                             |   
    |     *****      **                     * *                                      **                             |   
    |     *****      **                     * *                                      **                             |   
    |     *****      **                     * *                                      **                             |   
  3 +**** ***** **** ** **  **  **          * *                                  **  **              ***          +-+   
    |**** ***** ** * ** **  **  **          * *                                  **  **              * *            |   
    |**** ***** ** * ** **  **  **          * *                                  **  **              * *            |   
    |**** ***** ** * ** **  **  **          * *                                  **  **              * *            |   
  2 +**** ******** *******  ******    **    * *   **                   *****     ******   ******     * **         +-+   
    |**** ******** *******  ******    **    * *   **                   * * *     ** ***   * *  *     * **           |   
    |**** ******** *******  ******    **    * *   **                   * * *     ** ***   * *  *     * **           |   
    |**** ******** *******  ******    **    * *   **                   * * *     ** ***   * *  *     * **           |   
    +**** ******** *******+ ******    **    * *  +**        +          * * *     ** ***   * * +*     * **+          +   
  1 +****-****************+-******----**----***--+**--------+----------*****-----******---******-----****+--------+-+   
    0         0.1        0.2         0.3        0.4        0.5        0.6        0.7         0.8        0.9         1   
                                                    % of missing data                                                   

The 85% cutoff would be 0.775686
Would you like to set a different cutoff, yes or no
no

Excluding individuals in 'exclude' list
After filtering, kept 106 out of 122 Individuals
Outputting VCF file...
After filtering, kept 142840 out of a possible 142840 Sites
```

```
vcftools --vcf DP3g95maf05.recode.vcf --minDP 5 --maf 0.01 --recode --recode-INFO-all --out raw.g5mac3dp3
```
After filtering, kept 107565 out of a possible 142840 Sites

```
./pop_missing_filter.sh raw.g5mac3dp3.recode.vcf popmap 0.9 1 TRSdp5MIp25
```
After filtering, kept 21044 out of a possible 107565 Sites

```
./dDocent_filters TRSdp5MIp25.recode.vcf  TRSdp5MIp25g9dd

Number of sites filtered based on allele balance at heterozygous loci, locus quality, and mapping quality / Depth
 12811 of 21044

Are reads expected to overlap?  In other words, is fragment size less than 2X the read length?  Enter yes or no.
Number of additional sites filtered based on overlapping forward and reverse reads
2415 of 8233

Is this from a mixture of SE and PE libraries? Enter yes or no.
no

Number of additional sites filtered based on properly paired status
51 of 5818

Number of sites filtered based on high depth and lower than 2*DEPTH quality score
1893 of 5767



                                        Histogram of mean depth per site                                           
                                                                                                                    180+-+-+---+----+---+---+---+----+---+---+---+----+---+---+---+---+----+---+---+---+----+---+---+---+----+---+-+-*****   +    +   +   +   +    +   +   +   +    +   +   +   +   +    +   +   +   +    +   +   +   +    +   +   +   
  *   *                                               'meandepthpersite' using (bin($1,binwidth)):(1.0) ******|   
160 *-+ * +-+ *                                                                                               |   
140 *-+ *                                                                                          +-+ *                                                                                             |   
  *   *****                                                                                                     |   
120 *-+ *   *                                                                                                   +-+   
  *   *   *                                                                                                     |   
100 *-+ *   *                                                                                                   +-+   
  *   *   ******                                                                                                |   
  *   *   *    *****                                                                                            |   
80 *-+ *   *    *   *                                                                                          +-+   
  *   *   *    *   *                                                                                            |   
60 *-+ *   *    *   *********                                                                                  +-+   
  *   *   *    *   *   *   *                                                                                    |   
  *   *   *    *   *   *   *                                                                                    |   
40 *-+ *   *    *   *   *   *                                     ******                                       +-+   
  *   *   *    *   *   *  ***************************           *    *                                         |   
20 *-+ *   *    *   *   *   *    *   *   *   *    *   *************    *****       **********                  +-+   
  *   *   *    *   *   *   *    *   *   *   *    *   *   *   *   *    *   *****   *    *   *********        *****   
  *   *   *    *   *   *   *    *   *   *   *    *   *   *   *   *    *   *   *****    *   *   *   **********   *   
0 ***************************************************************************************************************   
  10  11  12   13  14  15  16   17  18  19  20   21  22  23  24  25   26  27  28  29   30  31  32  33   34  35  36  
                                                  Mean Depth                                                      

If distrubtion looks normal, a 1.645 sigma cutoff (~90% of the data) would be 4136.80545
The 95% cutoff would be 33
Would you like to use a different maximum mean depth cutoff than 33, yes or no
no
Number of sites filtered based on maximum mean depth
362 of 5767

Number of sites filtered based on within locus depth mismatch
18 of 5405

Total number of sites filtered
15657 of 21044

Remaining sites
**5387**

```

```
vcftools --vcf TRSdp5MIp25g9dd.FIL.recode.vcf --maf 0.05 --recode --recode-INFO-all --out DP3g95maf05d
```
After filtering, kept 3420 out of a possible 5387 Sites

```
rad_haplotyper.pl -v SNP.DP3gdp5MIp25g9HWE.recode.vcf  -x 10 -mp 2 -r reference.fasta
[evelyn-takyi@n038 Aurita_Final2]$ cp stats.out stats.out.HF
(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ mawk ...'/Missi/' stats.out.HF | mawk '$9 > 30' > HF.missing
(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ mawk ...'/para/' stats.out.HF > HF.para
(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ ./remove.bad.hap.loci.sh HF.loci.tofilter SNP.DP3gdp5MIp25g9HWE.recode.vcf  
(rad_haplotyper_env) [evelyn-takyi@n038 Aurita_Final2]$ mawk '!/#/' SNP.DP3gdp5MIp25g9HWE.filtered.vcf | wc -l
1479
```
```
vcfallelicprimitives -k -g SNP.DP3gdp5MIp25g9HWE.filtered.vcf| sed 's:\.|\.:\.\/\.:g' > TRSdp5MIp25g9.prim
vcftools --vcf TRSdp5MIp25g9.prim --recode-INFO-all --recode --out SNP.DP3g95maf05d --remove-indels
```
After filtering, kept 1592 out of a possible 1718 Sites

```
vcftools --vcf SNP.DP3g95maf05d.recode.vcf  --recode-INFO-all --max-alleles 2 --out SNP.2a --recode
```
After filtering, kept 1562 out of a possible 1592 Sites

##### Use PGDSpider to convert SNPs to BayeScan input file.
##### BayeScan2.1_linux64bits SNP1.BS -thin 100 -nbp 30
##### Create a VCF file of SNPs of outliers with a false discovery rate of 10%

```
./VCFtoOutlierOnly.sh SNPM.2a.recode.vcf  SNPMA_fst.txt  0.1 SNPMAdp  

Parameters as interpreted:
	--vcf SNPM.2a.recode.vcf
	--recode-INFO-all
	--out SNPMAdp.outlieronly
	--positions Outlier.list
	--recode

After filtering, kept 106 out of 106 Individuals
After filtering, kept 3 out of a possible 1562 Sites


VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf SNPM.2a.recode.vcf
	--exclude-positions Outlier.list
	--recode-INFO-all
	--out SNPMAdp.neutralonly
	--recode

After filtering, kept 106 out of 106 Individuals
After filtering, kept 1559 out of a possible 1562 Sites
```

### Analysis in R
#### Basic statistics

| Population | Private Alleles | Sites | Polymorphic Sites | % Polymorphic Loci | Obs Het | Exp Het | nucleotide diversity | Fis     |
|------------|-----------------|-------|-------------------|--------------------|---------|---------|----------------------|---------|
| Benin      | 9               | 1559  | 954               | 61.1931            | 0.266   | 0.1927  | 0.2029               | -0.1339 |
| Guinea     | 2               | 1559  | 713               | 45.7344            | 0.2464  | 0.173   | 0.2035               | -0.0803 |
| Mauritania | 5               | 1559  | 902               | 57.8576            | 0.2536  | 0.184   | 0.1974               | -0.1189 |
| Nigeria    | 447             | 1559  | 1013              | 64.9775            | 0.2212  | 0.2501  | 0.304                | 0.1936  |
| Senegal    | 4               | 1559  | 867               | 55.6126            | 0.2564  | 0.1828  | 0.1974               | -0.1227 |
| Togo       | 2               | 1559  | 885               | 56.7672            | 0.2616  | 0.1883  | 0.2016               | -0.1262 |

#### The overall FST and FIS values were calculated using the diffcalc function in the genepop package in R
```test_result <- diffCalc(infile = "SNPmadp.neutralonly.recode.gen", outfile = "madresults",fst = TRUE, pairwise = TRUE, bs_locus = TRUE, bs_pairwise = TRUE, boots = 1000, para = TRUE)
```
overall FST  =  0.1931	 CI (0.0858 - 0.3720)
overall FIS  = -0.1857	CI (-0.4258	- -0.0848)


#### Pairwise FST values calculated using the program Genodive
 Values of Fst for all pairs of populations is significant at (p<0.05)

 |            | Benin  | Guinea | Mauritania | Nigeria | Senegal |
 |------------|--------|--------|------------|---------|---------|
 | Benin      |        |        |            |         |         |
 | Guinea     | -0.003 |        |            |         |         |
 | Mauritania | 0      | 0      |            |         |         |
 | Nigeria    | 0.392  | 0.339  | 0.381      |         |         |
 | Senegal    | 0.002  | 0.006  | -0.002     | 0.368   |         |
 | Togo       | -0.005 | 0.016  | 0.007      | 0.374   | 0.009   |

p-values for all pairs of populations

  |            | Benin  | Guinea | Mauritania | Nigeria | Senegal |
  |------------|--------|--------|------------|---------|---------|
  | Benin      |        |        |            |         |         |
  | Guinea     | 0.562  |        |            |         |         |
  | Mauritania | 0.520  | 0.021  |            |         |         |
  | Nigeria    | 0.003  | 0.573  | 0.003      |         |         |
  | Senegal    | 0.354  | 0.240  | 0.652      | 0.003   |         |
  | Togo       | 0.818  | 0.037  | 0.092      | 0.003   | 0.061   |



### Neighbor joining tree
 ```{r}
 bov_pop <- genind2genpop(genindB)
 set.seed(20150428)
 pop_tree <- aboot(bov_pop, sample = 1000,
                            cutoff = 50)
```                            
![plot1](https://github.com/evelyn-takyi/Sardinella-project/blob/master/All%20plot/jessy7.png)


### DAPC

```set.seed(20160308) # Setting a seed for a consistent result
grp <- find.clusters(genindB, max.n.clust = 10, n.pca = 55, choose.n.clust = TRUE)
names(grp)
grp$grp
dapc1 <- dapc(genindB, grp$grp, n.pca = 10, n.da = 2)
scatter(dapc1) # plot of the group
```

![Dapc_BIC](https://github.com/evelyn-takyi/Sardinella-project/blob/master/BIC_MAD.png)

### structure-like plot using Posterior probability from dapc function
```
dapcB.results <- as.data.frame(dapc1$posterior)
dapcB.results$pop <- pop(genindB)
dapcB.results$indNames <- rownames(dapcB.results)
library(reshape2)
dapcB.results <- melt(dapcB.results)
colnames(dapcB.results) <- c("Original_Pop","Sample","Assigned_Pop","Posterior_membership_probability")

p <- ggplot(dapcB.results, aes(x=Sample,y=Posterior_membership_probability, fill=Assigned_Pop))
p <- p + geom_bar(stat='identity')
p <- p + facet_grid(~Original_Pop, scales = "free")
p <- p + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 5))
p

calculate mean per population
grouped_data <- group_by(dapcB.results, Original_Pop, Assigned_Pop)
data_means <- summarise(grouped_data, mean=mean(Posterior_membership_probability))
# plot means for each original pop colored by assigned pop
pieB <- ggplot(data_means, aes(x=Original_Pop,y=mean, fill=Assigned_Pop))
pieB <- pieB + geom_bar(stat='identity') + cp
pieB <- pieB + facet_grid(~Original_Pop, scales = "free") + theme(axis.text=element_blank(), axis.ticks=element_blank(), panel.grid = element_blank(), strip.background = element_blank())
pieB

cp <- coord_polar(theta = "y")
cp$is_free <- function() TRUE
```
![Plot3](https://github.com/evelyn-takyi/Sardinella-project/blob/master/All%20plot/jessy8.png)


### Analysis without Nigerian populatiion
#### Basic Statistic

| Population | Private Alleles | Sites | Polymorphic Sites | % Polymorphic Loci | Obs Het | Exp Het | nucleotide diversity | Fis     |
|------------|-----------------|-------|-------------------|--------------------|---------|---------|----------------------|---------|
| Benin      | 22              | 1112  | 954               | 85.7               | 0.372   | 0.270   | 0.284               | -0.187   |
| Guinea     | 2               | 1112  | 713               | 64.1               | 0.345   | 0.242   | 0.285               | -0.113   |
| Mauritania | 12              | 1112  | 902               | 81.1               | 0.355   | 0.258   | 0.276               | -0.167   |
| Senegal    | 24              | 1112  | 867               | 77.96              | 0.359   | 0.256   | 0.276               | -0.172   |
| Togo       | 17              | 1112  | 885               | 79.58              | 0.367   | 0.263   | 0.282               | -0.176   |


#### The overall FST and FIS values were calculated using the diffcalc function in the genepop package in R
```
wc(genindNM)
```
$FST
[1] 0.001773188

$FIS
[1] -0.3120278



#### Pairwise FST values calculated using the program Genodive
 Values of Fst for all pairs of populations is significant at (p<0.05)

 |            | Benin  | Guinea | Mauritania | Senegal |
 |------------|--------|--------|------------|---------|
 | Benin      |        |        |            |         |         
 | Guinea     | -0.003 |        |            |         |         
 | Mauritania | 0      | 0      |            |         |         
 | Senegal    | 0.002  | 0.006  | -0.002     |         |       
 | Togo       | -0.005 | 0.016  | 0.007      | 0.009   |

p-values for all pairs of populations

  |            | Benin  | Guinea | Mauritania | Senegal |
  |------------|--------|--------|------------|---------|
  | Benin      |        |        |            |         |        
  | Guinea     | 0.532  |        |            |         |        
  | Mauritania | 0.490  | 0.627  |            |         |        
  | Senegal    | 0.321  | 0.227  | 0.641      |         |
  | Togo       | 0.823  | 0.031  | 0.08       | 0.061   |

### Neighbor joining tree
   ```{r}
   bov_pop <- genind2genpop(genindB)
   set.seed(20150428)
   pop_tree <- aboot(bov_pop, sample = 1000,
                              cutoff = 50)
   ```
![Plot3](https://github.com/evelyn-takyi/Sardinella-project/blob/master/All%20plot/jessy10.png)

### structure-like plot using Posterior probability from dapc function

![Plot3](https://github.com/evelyn-takyi/Sardinella-project/blob/master/All%20plot/jessy11.png)
